tilelang/language/tir/ir.py

```
@@ -307,3 +307,73 @@ def wrapped(*args, **kwargs):
+# translates the kwargs into entries on the CastNode annotations map:
+#
+#   round=<str>  -> "round" (StringImm)
+#   sat=False    -> "sat"   (IntImm bool 0; sat=True is the default and
+#                            emits no annotation)
+#   rbits=<expr> -> "rbits" (PrimExpr; required when round="rs")
+# ---------------------------------------------------------------------------
+
+_VALID_CAST_ROUNDING_MODES = {"", "rn", "rz", "rp", "rm", "rs"}
+
+
+def cast(value, dtype, round: str = "", sat: bool = True, rbits=None, span=None):
```

Ruff (0.15.13)

[error] Function argument `round` is shadowing a Python builtin

(A002)